types/n3/index.d.ts

```diff
@@ -16,91 +16,119 @@ export interface Prefixes<I = RDF.NamedNode> {
+    static subclass(type: any): void;
 }
 
-export class Literal extends Term implements RDF.Literal {
+export class Literal implements RDF.Literal {
     static readonly langStringDatatype: NamedNode;
     termType: "Literal";
     value: string;
+    id: string;
+    toJSON(): {};
+    equals(other: RDF.Term): boolean;
+    static subclass(type: any): void;
     language: string;
-    datatype: RDF.NamedNode;
+    datatype: NamedNode;
     datatypeString: string;
     constructor(id: string);
 }
 
-export class DefaultGraph extends Term implements RDF.DefaultGraph {
+export class DefaultGraph implements RDF.DefaultGraph {
     termType: "DefaultGraph";
     value: "";
     constructor();
+    id: string;
+    toJSON(): {};
+    equals(other: RDF.Term): boolean;
+    static subclass(type: any): void;
+}
+
+export type Quad_Subject = NamedNode | BlankNode | Variable;
+export type Quad_Predicate = NamedNode | Variable;
+export type Quad_Object = NamedNode | Literal | BlankNode | Variable;
+export type Quad_Graph = DefaultGraph | NamedNode | BlankNode | Variable;
+
+export class BaseQuad implements RDF.BaseQuad {
+    constructor(subject: Term, predicate: Term, object: Term, graph?: Term);
+    subject: Term;
+    predicate: Term;
+    object: Term;
+    graph: Term;
+    equals(other: RDF.BaseQuad): boolean;
+    toJSON(): string;
 }
 
-export class Quad implements RDF.Quad {
-    constructor(subject: RDF.Term, predicate: RDF.Term, object: RDF.Term, graph?: RDF.Term);
-    subject: RDF.Term;
-    predicate: RDF.Term;
-    object: RDF.Term;
-    graph: RDF.Term;
-    equals(other: RDF.Quad): boolean;
+export class Quad extends BaseQuad implements RDF.Quad {
+    constructor(subject: Term, predicate: Term, object: Term, graph?: Term);
+    subject: Quad_Subject;
+    predicate: Quad_Predicate;
+    object: Quad_Object;
+    graph: Quad_Graph;
+    equals(other: RDF.BaseQuad): boolean;
     toJSON(): string;
 }
 
 export class Triple extends Quad implements RDF.Triple {}
 
 export namespace DataFactory {
-    function namedNode(value: string): RDF.NamedNode;
-    function blankNode(value?: string): RDF.BlankNode;
-    function literal(value: string | number, languageOrDatatype?: string | RDF.NamedNode): RDF.Literal;
-    function variable(value: string): RDF.Variable;
-    function defaultGraph(): RDF.DefaultGraph;
-    function triple(subject: RDF.Term, predicate: RDF.Term, object: RDF.Term): RDF.Quad;
-    function quad(subject: RDF.Term, predicate: RDF.Term, object: RDF.Term, graph?: RDF.Term): RDF.Quad;
+    function namedNode(value: string): NamedNode;
+    function blankNode(value?: string): BlankNode;
+    function literal(value: string | number, languageOrDatatype?: string | RDF.NamedNode): Literal;
+    function variable(value: string): Variable;
+    function defaultGraph(): DefaultGraph;
+    function quad(subject: RDF.Quad_Subject, predicate: RDF.Quad_Predicate, object: RDF.Quad_Object, graph?: Quad_Graph): Quad;
```

Comment:
The `graph` param should be `RDF. Quad_Graph`.

Reply:
Resolved :+1: